### ISSUE TITLE
tracking protection - why exactly is it disabled?

### ISSUE THREAD

**Issue by RoxKilly** [ghacksuserjs/ghacks-user.js#102]
The implementation guide explains:

> Tracking Protection (TP) and Safe Browsing (SB) are turned off (section 0400) -- We think you can do much better than this (wider scope, no over-reach / censorship)

Could someone please elaborate? Is the block because the browser has to connect to a remote server to download blocklists? @Thorin-Oakenpants do you operate under the assumption that this `user.js` must be used in conjunction with uBlock Origin? If this is meant for both uBO users and non-users, why disable TP?

Setting that aside, let me make the case for TP even for a uBO user like myself: For the vast majority of webpages, TP never plays a role because uBO blocks requests before they get to the TP code (see the last comment from link 3 below). So there is no additional burden on the browser and I don't see an additional privacy exposure (beyond the blocklist downloads).

In some cases, default uBO filter lists and settings let something through the cracks and TP actually catches it (eg: enable Tracking Protection and open [this page](https://smile.amazon.com/gp/help/customer/display.html/) as of May 4 2017). This is usually a tracking image of some sort. In those cases I'm glad to have TP on.

### For Reference
1. [How TP works](https://feeding.cloud.geek.nz/posts/how-tracking-protection-works-in-firefox/)
2. [TP wiki](https://wiki.mozilla.org/Security/Tracking_protection)
3. [TP authors' post](https://blog.mozilla.org/nnethercote/2014/11/12/quantifying-the-effects-of-firefoxs-tracking-protection/) -- the comments below the post yield a lot of useful info; for instance although TP uses the SafeBrowsing protocol, there is no interaction with Google..

**Comment by earthlng** [comment]
> In some cases, my uBO filter lists and settings let something through the cracks and TP actually catches it.

afaik TP uses the same lists (Disconnect) that are already available in uBO. Either your uBO lists are not updated or I'd love to see an example of those cases you're taking about.

**Comment by Thorin-Oakenpants** [comment]
This js started as my own private project with my choices. It went on ghacks and the general feedback was that everyone/most liked this - part of the original spec as keeping FF "quiet" (which as a term from a wilders security thread) - section 0400 was created to split out FF's SB & TP stuff. It was left activated all thru my versions on ghacks.

I have no worries myself with SB re security/privacy (except the real time binary checks - how this checks may have changed since francois' blog article listed in the 0400 section). My only personal gripe with it would be over-reach/censorship - eg blocks of TPB or whatnot (see torrentfreak for numerous articles). As for TP, I have read numerous accounts of this being a cause for breakage - so to me personally, I prefer to have one less item to deal with.

When we moved to github I was concerned that I did not want to put users at risk, despite all the warnings (and it took me time to write the wiki implementation, readme etc). Earthlng and myself almost came to blows when I suggested allowing TP/SB in the template. My point was I don't want to put anyone at risk and in order to cater for the lower common denominator .... , and earthlng was, well, like, this is aimed at people with some knowledge (don't want to put words in his mouth) ... the whole exchange here on GH was removed.

AFAIK - The ONLY reason to enable these is for people who are not prepared to use uBo, which is not that hard to use.

**Comment by earthlng** [comment]
> the whole exchange here on GH was removed

but worried about censorship - oh the irony xD

**Comment by Thorin-Oakenpants** [comment]
Self removal of own content is not censorship :)

**Comment by earthlng** [comment]
no point, just kidding :)

**Comment by RoxKilly** [comment]
@earthlng wrote:

> afaik TP uses the same lists (Disconnect) that are already available in uBO. Either your uBO lists are not updated or I'd love to see an example of those cases you're taking about.

You probably have a point there. The Disconnect lists were not selected in my uBO filters. Is Tracking Protection (TP) list limited to Disconnect? [This link](https://wiki.mozilla.org/Security/Tracking_protection#Lists) suggests otherwise.  The next time I come across such an example, I'll try to remember to post it here; it happens rarely though.

If you guys have made the conscious decision to not cater to people who don't also use uBO then I understand the decision better. If this `user.js` is meant to be privacy-enhancing on its own, then I don't understand why TP would be off by default: its upside for privacy is so huge for those without uBO or something similar

**Comment by Thorin-Oakenpants** [comment]
> If you guys have made the conscious decision to not cater to people who don't also use uBO then I understand the decision better. If this user.js is meant to be privacy-enhancing on its own

So did you not read the Implementation wiki page with the SUPER UBER IMPORTANT section, or the express warning in the readme in the user.js .. or the readme which states "(with add-ons)".  :) I think we've made it painfully aware that we expect users to either remove section 0400 or use uBo.

That said .. I think things will be a little clearer for end users when we provide a "relaxed" or "lite" branch, in which TP/SB is enabled

**Comment by RoxKilly** [comment]
@Thorin-Oakenpants  I did read the Implementation wiki. I even quote it in the OP. But the only thing it says about uBO is:

> There is nothing wrong with running TB and SB as well as uBlock Origin

This doesn't get across that people who use this without uBO should not expect privacy. If that is your position, consider making it more obvious.

I tried to make the case regardless of whether uBO is used; please take anothing look at my previous post. Thanks for reading.

**Comment by earthlng** [comment]
> Is Tracking Protection (TP) list limited to Disconnect?

it seems so:
from https://feeding.cloud.geek.nz/posts/how-tracking-protection-works-in-firefox/

> The blocklist is created by Disconnect according to their definition of tracking.

You can choose between recommended and strict, and that changes `urlclassifier.trackingTable`
```
strict:                             test-track-simple,base-track-digest256,content-track-digest256
default (recommended): test-track-simple,base-track-digest256
```

If you look at [this](https://github.com/mozilla-services/shavar-list-creation-config/blob/master/prod.ini)
both `base-track-digest256` + `content-track-digest256` are based on [this](https://raw.githubusercontent.com/mozilla-services/shavar-prod-lists/master/disconnect-blacklist.json), both coming from disconnect.

**Comment by Thorin-Oakenpants** [comment]
I'm open to suggestions. When we got here to GH, I seriously, almost, despite earthlng's stance, enforced SB/TP on.

> There is nothing wrong with running TB and SB as well as uBlock Origin

Well, it's short bullet points, not an essay. There are considerations, but the "nothing wrong" is about privacy/security, and exactly on point. I think we can agree on that. I'm not arguing against you.

> This doesn't get across that people who use this without uBO should not expect privacy. If that is your position, consider making it more obvious.

How? What exactly should I be writing here? I've never inferred that non-uBo users should not have TP/SB on and thus don't deserve privacy etc. I don't follow.

**Comment by earthlng** [comment]
> Thanks for reading.

Pants can sometimes be a bit salty. Please don't let this spook you away from here. You provide quality posts and comments, and I'd hate to see you leave.

**Comment by Thorin-Oakenpants** [comment]
edited wiki

original
```
  * We think you can do much better than this (wider scope, no over-reach / censorship)
```
to
```
  * We think you can do much better than this (wider scope, no over-reach / censorship) by using uBlock Origin
```

**Comment by RoxKilly** [comment]
@earthlng Thanks for the encouragement. I'm in no danger of leaving; I was being sincere with my thanks. I'm curious: what do you think about following assessment:

* for people without a uBO-like tool, having TP enabled is a major privacy win. Disabling TP exposes them to tracking

* for people with uBO's Disconnect list loaded, TP enabled offers no added benefit, but no downside either because that code path is never traveled

* so enabling TP is a win for some, and makes no difference to others.

Gets back to my original question: what is the benefit of disabling it?

**Comment by Thorin-Oakenpants** [comment]
> I tried to make the case for Tracking Protection regardless of whether uBO is used

So did I, months ago, with earthlng. In the end we decided (and we were basically the only ones here, and have been building this together for over a year via ghacks) to make sure the warnings/info was everywhere possible. That's not to say we can't improve that - as per your interpretation of the wiki entry bullet point (an fair enough too - fresh eyes are good). That also not to say that we can;lt reverse the decision.

**Comment by Thorin-Oakenpants** [comment]
I keep bundling TP/SB together in my comments. Sorry for throwing that into this discussion.

Sticking to TP only - I have no issues with this. I do not care about the connections (a lot of ghacks users liked the "quiet" part - Just Me is one, he even kills kinto), they are secure etc. I have no issues with enforcing it since it has no conflicts or side-effects. And as a fallback (if we call it that) for users who don;t use uBo (who who don't read), then it is all good. These are the same arguments I made months ago.

**Comment by Thorin-Oakenpants** [comment]
I guess we better also fix the two disabled kinto prefs

**Comment by Thorin-Oakenpants** [comment]
What is the pref for which TB list to use. What setting do we give it?

**Comment by Thorin-Oakenpants** [comment]
We'll also have to make sure the prefs used to get the URLs are reset as well.

Feel free to make a PR

**Comment by earthlng** [comment]
@RoxKilly, I see what you mean but I just can't imagine that someone would use something like this user.js but not use uBO. It can be used simply as a better version of ABP. You don't even have to use the more advanced features and IMO it's really easy to setup and use. Not to mention the amazing Element picker.

> so enabling TP is a win for some, and makes no difference to others

it makes the slight difference that it downloads the blocklists (every hour?) basically for no reason at all.

And as for the sentence you two are arguing about ....

> There is nothing wrong with running TB and SB as well as uBlock Origin

I'd say the important part is **as well as uBlock Origin**

IMHO uBO is an absolute must-have addon, and I'd rather we make that abundantly clear instead of enabling TP (and/or SB)

**Comment by earthlng** [comment]
> What is the pref for which TB list to use. 

Do you not read, SaltyPants? ;)
https://github.com/ghacksuserjs/ghacks-user.js/issues/102#issuecomment-298346502

When changing it in the options it needs to restart FF btw

**Comment by RoxKilly** [comment]
> What is the pref for which TB [sic] list to use. What setting do we give it?

I don't yet know which preference determines whether the basic or strict protection is enabled. But I know how to change it in the UI, if that helps: With `privacy.trackingprotection.ui.enabled = true`, go to `about:preferences#privacy` and click the button labeled "*Change Block List*" to make the change. My advice is to leave it as the default ("*basic list*", same as that in the uBO filters)

> We'll also have to make sure the prefs used to get the URLs are reset as well

To get tracking protection working in my browser all I had to do was comment out `0410d` and set `0420` prefs to `true`.  I did *not* need to fix the URLs in `0410e` and `0410f`

**Comment by earthlng** [comment]
> I don't yet know which preference determines whether the basic or strict protection list is used. 

seriously? you too? why am I even posting here if nobody reads my shit xD

**Comment by RoxKilly** [comment]
@earthlng LOL

The download itself doesn't occur every hour. A check is made for whether there is an update every hour:

> Every hour, Firefox requests updates from shavar.services.mozilla.com. If new data is available, then the whole list is downloaded again. Otherwise, all it receives in return is an empty 204 response. ([source](https://feeding.cloud.geek.nz/posts/how-tracking-protection-works-in-firefox/))

**Comment by Thorin-Oakenpants** [comment]
Yeah I did . My point was which list to use, you'd have to also add the pref and number and description. Like I said, create a PR - make sure to fix the URLs prefs and make sure everything works ..

and as I type @RoxKilly confirms some stuff. Yes 0410e is realtime binary checks and 0410f is reporting URLs. Not talking about those.

```js
/* 0410d: disable Mozilla safebrowsing downloads, updates
 * [NOTE] These two prefs are also used for Tracking Protection (see 0420) ***/
user_pref("browser.safebrowsing.provider.mozilla.gethashURL", ""); // resolves hash conflicts
user_pref("browser.safebrowsing.provider.mozilla.updateURL", ""); // update FF lists
```

Did you need to reset these two?

===
Yup, we all know about the settings in the UI - we can add in a [SETTING] line so people know where to look for that. The user.js though needs prefs. Earthlng provided the answer `urlclassifier.trackingTable`. Yes earthlng - I read your stuff.

**Comment by RoxKilly** [comment]
@Thorin-Oakenpants I simply commented out those two lines. My copy of `user.js` has:
```
   // user_pref("browser.safebrowsing.provider.mozilla.gethashURL", "");
   // user_pref("browser.safebrowsing.provider.mozilla.updateURL", ""); 
```

In `about:config`, these are the values I see:
```
browser.safebrowsing.provider.mozilla.gethashURL = https://shavar.services.mozilla.com/gethash?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2
browser.safebrowsing.provider.mozilla.updateURL = https://shavar.services.mozilla.com/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2
```

I've also just confirmed that Tracking Protection is working by disabling uBO, then loading nytimes.com

**Comment by Thorin-Oakenpants** [comment]
Let's not get distract-- .. ooh, look, a beer

- hourly connections are not a consideration
- whether it pulls anything down or not is irrelevant
- security is not an issue
- privacy is not an issue

We need a PR that enables TP, makes sure that the required URLs are not disabled, and we need to add in (and decide) the TP list level (I suggest listing the strict mode as inactive with a description such as enforce strict mode) as say pref 0422

I need to work this out (but I think someone else could do it). First of all
- Options>Privacy>Using Tracking Protection: "only in private windows" vs "always" - what pref is this again?
- Blocklist options - if we use strict/simple - this is across all windows right? normal & PB?

**Comment by earthlng** [comment]
> Yes earthlng - I read your stuff.

THANK YOU, ffs :)

> Like I said, create a PR

Hell no! I like the way it is right now. Beginning to hate @RoxKilly - now you're 2 against 1 - that sucks! xD

```js
/* 0420: disable Tracking Protection (TP)
 * There SHOULD be NO privacy concerns here, but we strongly recommend to use uBlock Origin instead,
 * which offers more comprehensive as well as specialized lists. ... ***/
```

> **strongly recommend to use uBlock Origin instead**

**Comment by Thorin-Oakenpants** [comment]
@RoxKilly I thought so. I mean I typed that whole 0400 section like a year ago, with excellent notation, but you never know when urls/prefs/code changes. Good to know that just those 2 urls (0410d) were needed as per notation.

@earthlng It's a template, like we keep telling people. We can still leave all the notation in, we just comment out 4 prefs and add in one commented pref for info (about using strict mode).

**Comment by RoxKilly** [comment]
@Thorin-Oakenpants wrote:
> Options>Privacy>Using Tracking Protection: "only in private windows" vs "always" - what pref is this again?

"TP only in private windows" is:
```
privacy.trackingprotection.enabled = false
privacy.trackingprotection.pbmode.enabled = true
```
Those are the default settings by the way

"TP always" is:
```
privacy.trackingprotection.enabled = true
privacy.trackingprotection.pbmode.enabled is irrelevant (I just checked)
```

> Blocklist options - if we use strict/simple - this is across all windows right? normal & PB?

I don't know for a fact but I don't see why it would be any other way.

**Comment by Thorin-Oakenpants** [comment]
>  now you're 2 against 1 

It's called a majority :) BUT, this is not a done deal. I want more feedback from other users. I see zero downsides to enabling TP. Yes it's redundant for "us", but it's responsible fallback for "newbies" (for lack of a better word).

This requires work. Extra prefs, changing readme info, changing the wiki, and do we have that pref for using TP in all windows vs none vs PB mode only (another pref to add)? It will not be done today - like I said I want more feedback.

**Comment by earthlng** [comment]
Ok, I'll let you two wrap each other in TP while I go watch some funny cat videos.
If you're seriously gonna do this right now, you may want to consider dealing with these too:
```
pref("privacy.trackingprotection.annotate_channels", false);
pref("privacy.trackingprotection.lower_network_priority", false);
```

happy TP-ing you two, laterz

**Comment by Thorin-Oakenpants** [comment]
I doubt anything will happen for a week - I want more than just three people's feedback/ideas :) That's a lot of cat videos @earthlng xD

**Comment by Thorin-Oakenpants** [comment]
```js
pref("privacy.trackingprotection.annotate_channels", false);
pref("privacy.trackingprotection.lower_network_priority", false);
```
are both not in the user.js so don't comer into play (yet)

**Comment by Atavic** [comment]
Both set as  False here, with the URL removed as [Roxkilly](https://github.com/ghacksuserjs/ghacks-user.js/issues/102#issuecomment-298356290).

Potential amazon tracking: https://www.robtex.com/dns-lookup/shavar.services.mozilla.com

**Comment by Thorin-Oakenpants** [comment]
@Atavic said

> If enabled, leave it on the relaxed branch only.
Reason: Anti-fingerprinting.

Well vote then :)  "potential" fingerprinting? FP'ing? really? I'll have to read it, I spose. I couldn't possibly imagine how TP's list (which is also in uBo) changes FP'ing. I'm sure Mozilla's distribution method is fine. Time for a coffee and read then

**Comment by Thorin-Oakenpants** [comment]
@Atavic I don't follow. How does this FP anyone - the hourly check leaks a specific IP which lowers entropy?

**Comment by Atavic** [comment]
Browser retrieves the hash for updates from:
`https://shavar.services.mozilla.com/gethash?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2`
where SAFEBROWSING_ID is the same as [Google API_KEY](https://developers.google.com/safe-browsing/v4/lookup-api).

By looking at robtex [here](https://github.com/ghacksuserjs/ghacks-user.js/issues/102#issuecomment-298365481), you see the service is hosted at Amazon.

**Comment by Thorin-Oakenpants** [comment]
Right .... so HOW does that FP you compared to say .. the other 500 million FF users?

**Comment by Atavic** [comment]
/*** 1200 HSTS

> HTTP Strict Transport Security (HSTS) with long duration deployed on this server.

[Source](https://www.ssllabs.com/ssltest/analyze.html?d=shavar.services.mozilla.com&s=52.39.210.199)

https://github.com/gorhill/uMatrix/issues/389
[hsts-tracking](https://www.ghacks.net/2015/10/16/how-to-prevent-hsts-tracking-in-firefox/)

**Comment by Thorin-Oakenpants** [comment]
@kaliostro2 said

> We have uBlock Origin , uMatrix and soon full-fledged containers with "Always open in this conteiner".

That's kind of the point. This covers users who DON'T have uBo and for those who do, leaving TP at defaults is not an issue security/privacy/ or clashing-wise

**Comment by Thorin-Oakenpants** [comment]
@Atavic ELI5 (explain it like I'm five) buddy. So it generates a HSTS supercookie? And its checked every hour - and the id stays the same? What's the big deal, I still don't get it. Are Mozilla creating a unique ID on you, storing it, and using it against you somehow?

**Comment by Atavic** [comment]
My problem is AmazonCDN hosting.

**Comment by crssi** [comment]
Huh... missed the whole party here. :)
Is there any final verdict? Since the English is not my mother language I have a bit trouble to follow up. ;)
What I like, if FF is doing the "protection", is the way FF shows it to the user as seen on http://itisatrap.org pages.... the red page with warning I mean.

To make it work, I have set those:
	/* 0410a */ `user_pref("browser.safebrowsing.malware.enabled", true);`
	/* 0410a */ `user_pref("browser.safebrowsing.phishing.enabled", true); `
	/* 0410d */ `user_pref("browser.safebrowsing.provider.mozilla.gethashURL", "https://shavar.services.mozilla.com/gethash?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2");`
	/* 0410d */ `user_pref("browser.safebrowsing.provider.mozilla.updateURL", "https://shavar.services.mozilla.com/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2");`

Does my comment make any sense?

**Comment by Thorin-Oakenpants** [comment]
> Is there any final verdict? Since the English is not my mother language I have a bit trouble to follow up. ;)

I've simplified it in #103 where you can vote on it. Keep discussions in here though if you can. No final decision has been made. I can see both sides and TBH, I haven't voted yet. I'm still intrigued why Atavic thinks this FP's him. So Amazon FPs people thru mozilla's TP lists via HSTS supercookies? I still don't get it.

**Comment by Thorin-Oakenpants** [comment]
@crssi : we're talking about Tracking Protection only. Your settings above are about Safe Browsing ( (except 0410d which is used by Tracking Protecting as well as Safe Browsing).

**Comment by crssi** [comment]
Thank you pants. What about SB?
I was holding those questions for the lite version, but @RoxKilly opened a pandora box before lite version. ;)

**Comment by Atavic** [comment]
So Mozilla hosts such services on Amazon servers. Will Amazon use the connections to track the browser?  They definitely could, as an API key is a unique value that is assigned to a user of the service.

https://github.com/Synzvato/decentraleyes/issues/99

**Comment by Thorin-Oakenpants** [comment]
> What about SB?

Oh jeeze crssi .. don't go there mate :) This is even more important for non-uBo users. This is the red warning pages for ya old old folks - malware, phishing, pups etc that you mentioned above. Personally, I do not want this, I am a big girl who can take care of herself. The same reasons for enabling SB apply as TP - to not put anyone at risk (but, we have tons of warnings). It does get a little more comlex in that there are real-time binary checks made by google, but if we ignore those, the rest is much the same. Leave it for a lite branch, or leave at defaults in this branch by deactivating them.

**Comment by crssi** [comment]
I will shut up now. :)

**Comment by fmarier** [comment]
> `pref("privacy.trackingprotection.annotate_channels", false);`
> `pref("privacy.trackingprotection.lower_network_priority", false);`

Note that enabling these won't do anything if `privacy.trackingprotection.enabled` is ON. These prefs exist to de-prioritize known trackers when they're not blocked outright. If you enable tracking protection, then the tracking resources don't need to be de-prioritized because they never load at all :)

> Browser retrieves the hash for updates from:
https://shavar.services.mozilla.com/gethash?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2
> where SAFEBROWSING_ID is the same as Google API_KEY.

As [documented here](https://wiki.mozilla.org/Security/Safe_Browsing#Prefs), `SAFEBROWSING_ID` is not the Google API Key. Instead, it gets replaced with the contents of `browser.safebrowsing.id` (which defaults to `navclient-auto-ffox`).

**Comment by RoxKilly** [comment]
@Atavic , the use of HSTS by itself does not mean fingerprinting. HSTS is actually a great security feature because it protects against a man-in-the-middle downgrade attacks by forcing the browser to connect only over a secure channel by default.

Using HSTS to fingerprint requires having the browser make many connections to many domains (subdomains usually) and testing whether the browser knows to connect to these domains over HTTPS by default. So unless we have evidence that the Tracking Protection server connection involves many domains with HSTS there can be no fingerprinting of that sort.

In addition, Firefox [doesn't save HSTS settings in Private Browsing](https://www.reddit.com/r/firefox/comments/32v9sr/privacy_how_to_prevent_hsts_tracking/) mode, so in the browser's default configuration (Tracking Protection used during Private Browsing) this fingerprinting technique wouldn't even work. Again, I think we need evidence that there is even fingerprinting going on in the first place for this to be a concern.

I can confirm that my `browser.safebrowsing.id` is simply set to `navclient-auto-ffox`, so there is no inherent fingerprinting either in the URL that connects to the blocklist server: 
```
browser.safebrowsing.provider.mozilla.gethashURL 
    => https://shavar.services.mozilla.com/gethash?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2
browser.safebrowsing.provider.mozilla.updateURL 

    => https://shavar.services.mozilla.com/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2
```

I'm having a hard time seeing how there is tracking going on.  Seems to me that at most, the Amazon hosted server `shavar.services.mozilla.com` will know that a Firefox browser at my IP is using the Tracking Protection blocklist. This is true in default Firefox installs though so it's hardly identifying.

**Comment by Gitoffthelawn** [comment]
I keep reading that uBo has the same default lists as TP.

Which lists are people talking about?  When I compare the Disconnect tracking list in uBo to the TP lists, they appear very different.

**Comment by earthlng** [comment]
> When I compare the Disconnect tracking list in uBo to the TP lists, they appear very different.

Indeed. No idea why that is. But TP thinks allowing google-analytics.com (among others) on google is totally fine. I can't take TP serious when it does things like that.
At least the [mozilla version](https://github.com/mozilla-services/shavar-prod-lists/blob/master/disconnect-entitylist.json) moved the questionable domains from "properties" to "resources" (compared to the [original](https://s3.amazonaws.com/lists.disconnect.me/entitylist.json)) but idk how much of a difference that makes.

**Comment by Atavic** [comment]
@Roxkilly Thanks for reminding me that Private Browsing mode allows to surpass the potential privacy problem.

**Comment by earthlng** [comment]
Is `browser.safebrowsing.provider.mozilla.gethashURL` even used for TP? I don't think so

**Comment by fmarier** [comment]
> Indeed. No idea why that is. But TP thinks allowing google-analytics.com (among others) on google is totally fine. I can't take TP serious when it does things like that.

TP is designed to block _third-party_ tracking. If you're visiting GMail for example, they are clearly able to track you regardless of whether or not you block `google-analytics.com`. They have you in their server logs and they can run all the JavaScript they want in your browser.

On the other hand, Google shouldn't have to know that you're visiting CNN so `google-analytics.com` will be blocked by TP there.

> Is `browser.safebrowsing.provider.mozilla.gethashURL` even used for TP? I don't think so.

Not currently. The TP list doesn't need it since it contains the full hashes, not the partial hashes like the other Safe Browsing lists.

**Comment by RoxKilly** [comment]
@earthlng wrote:
> But TP thinks allowing google-analytics.com (among others) on google is totally fine. I can't take TP serious when it does things like that...and googlesyndication.com was not blocked

This won't change how you feel but there is a good explanation for this.  The objective of TP is to reduce tracking without breaking sites. Where sites break, functionality supersedes privacy. Firefox (or Disconnect) may have discovered that in the wild, a lot of sites were breaking. Mozilla even wrote a special [article about the breakage](https://developer.mozilla.org/en-US/Firefox/Privacy/Tracking_Protection#What_does_this_mean_for_your_website) here. 

What uBO does -- and one reason it's such an amazing tool -- is use its [`redirect` syntax](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#redirect) to replace the tracking functions with a neutered one, so that sites won't break. For people without uBO & Co., Tracking Protection is a big deal and it [significantly reduces 3rd party exposure](https://blog.mozilla.org/nnethercote/2014/11/12/quantifying-the-effects-of-firefoxs-tracking-protection/); I wouldn't be so dismissive.

**Comment by earthlng** [comment]
@fmarier, thanks. But the google-analytics.com scripts are doing all (or most) of the tracking, don't they?

So how does that work exactly - ELI5 please - all the domains in "properties" can load whatever they want from the domains listed in "resources" ? (list [here](https://github.com/mozilla-services/shavar-prod-lists/blob/master/disconnect-entitylist.json))
I'm confused because you mentioned gmail but that is listed under "resources". Or can all domains in "properties" AND "resources" access all domains in both lists? Just out of curiosity.

@RoxKilly so you're trying to tell me that google.com would break if google-analytics.com would be blocked? Come on - even if that was the case it was deliberately created that way, exactly because, as I suspect, google-analytics.com scripts are google's main tracking feature.
I can use google.com with scripts blocked and uBO doing its thing in the background - no problems at all.

> This probably won't change how you feel

nope, but thanks for the link about breakage, and of course the rest of your comment too :)

**Comment by fmarier** [comment]
> But the google-analytics.com scripts are doing all (or most) of the tracking, don't they?

That's a question for Google :) They certainly have the technical capability to track their users in lots of other ways when they connect to their own servers.

I would assume that they have the ability to track me when I use their services regardless of the add-ons / privacy settings I have. The possible exception would be if I'm using the Tor Browser Bundle.

> all the domains in "properties" can load whatever they want from the domains listed in "resources" ?

Yes, that's essentially how it works.

> I'm confused because you mentioned gmail but that is listed under "resources". Or can all domains in "properties" AND "resources" access all domains in both lists?

GMail is confusing because the app is on `google.com` (which is in `properties`) and not `gmail.com`.

> so you're trying to tell me that google.com would break if google-analytics.com would be blocked?

For an example where things are quite broken, try blocking `twimg.com` all the time. Twitter is unusable without it. With the entity list, we can group `twitter.com` and `twimg.com` together (since they are the same company) and eliminate that breakage while blocking `twimg.com` as a third-party tracker on other sites.

**Comment by earthlng** [comment]
> That's a question for Google :)

Indeed :) I doubt however that they would be all too willing to let us know :(
"Don't be evil" my ass ;)

> They certainly have the technical capability to track their users in lots of other ways when they connect to their own servers.

No doubt about that. But that doesn't mean you have to make it easier for them. Think about it - if you had to design it, you would want to centralize it somewhere and collect and aggregate it, ideally all through one channel. IMO that channel is GA, at least to a large degree.

> Yes, that's essentially how it works.

Essentially? with exceptions? if ELI5 is too much to ask, ELI12 is fine too :)

> twimg.com

I have no problem with allowing domains used to host images for the "main" domains (twitter.com in this case)

I'm just throwing this out there, you don't have to reply to this, but come on - everyone knows nobody wants to piss off google & co because everyone wants their money. That's the only reason why they get special treatment. I assume Disconnect is "in on it" too. I don't blame you (mozilla!) - without money from somewhere you could not work on the awesome project you work on, and I for one am very happy that we have a very good alternative to Chrome. So Thanks for that and keep fighting the good fight :)

**Comment by Gitoffthelawn** [comment]
@earthlng What Disconnect list are you using in uBo to cover a superset of everything in TP?

**Comment by earthlng** [comment]
in uBO? none. I block all domain lists in uMatrix. Apart from the default ones I use these (with limited usefulness)
```
https://s3.amazonaws.com/lists.disconnect.me/simple_ad.txt‎ : 0 used out of 2'703
https://s3.amazonaws.com/lists.disconnect.me/simple_malvertising.txt‎ : 568 used out of 5'337
https://s3.amazonaws.com/lists.disconnect.me/simple_malware.txt‎ : 0 used out of 2'601
https://s3.amazonaws.com/lists.disconnect.me/simple_tracking.txt‎ : 0 used out of 34
```

**Comment by RoxKilly** [comment]
@earthlng  wrote:
> so you're trying to tell me that google.com would break if google-analytics.com would be blocked?

No I was referring to other sites that rely on google-analytics in a way that breaks the site if the analytics aren't loaded properly. Anyway I feel like we're going in circles. We're in general agreement (uBO > TP > nothing). I only jumped in because you initially hinted that TP was a joke. but I took exception to that.

**Comment by earthlng** [comment]
> No I was referring to other sites that rely on google-analytics in a way that breaks the site

were you though? because TP blocks GA on "other sites". You can't have it both ways, now you're basically saying the opposite of what you wrote above. You were trying to sell me the advantages of TP, member ;)
"TP is not a joke but it doesn't want to break sites"

**Comment by Thorin-Oakenpants** [comment]
I think it's pointless arguing TP vs uBo. They have different end goals and target audiences IMO

TP is clearly designed to take the complexity out of tracking and make it super easy for end-users - such as preconfiguring things such as twitter + twimg (these are thousands of examples of these - amazon images etc - so many separate domains for hosting various media, CDNs etc) - this also means no site breakage, because that would upset said non-tech-savy end-users and drive them away. This is Mozilla being responsible, they have to err on the side of caution. They did at least introduce it, which is more than can be said of other browsers' pathetic lack of concern for privacy - and FF provides switches to ramp it up (simple/strict) and expand it (all windows).

uBo is superior. I do not consider it complicated to use, but many might. So called eg twitter+twimg I had to configure myself (not sure on the default install config, but I am in deny-all mode except first party - PS i have NS first in deny all). uBo allows user control (more lists, choose what lists, custom rules, granular control) and more scope (all windows) etc.

**Comment by Gitoffthelawn** [comment]
In #103, the great @Thorin-Oakenpants wrote:
>TP uses the same list as uBo (note: there are two lists, simple+strict, default is simple and we would leave it at that but include the pref for info/enforcing strict)

What are you talking about?  What list?  Is uBo configured by default to use this list?  Are these lists always identical?

**Comment by Thorin-Oakenpants** [comment]
@Gitoffthelawn I've probably mixed that up with the SB lists. I'm not really concerned, uBo covers what TP does, so TP is redundant in that sense. I'll let someone else provide exact answers as to where the lists comes from, who provides/curates them etc. I thought Disconnect did TP and google SB.

**Comment by Gitoffthelawn** [comment]
@Thorin-Oakenpants No problem.  Although it's certainly likely, I haven't seen anything conclusive that uBo (especially out of the box) actually covers everything that TP covers.  Of course, uBo (or any content blocker) can block whatever TP blocks, but that requires a list in a compatible format.

**Comment by Thorin-Oakenpants** [comment]
chrome://ublock0/content/dashboard.html#3p-filters.html -> "Basic tracking list by Disconnect". I'm going to change browsers and see what is enabled by default on a new uBo install. Will report back

**Comment by RoxKilly** [comment]
uBO doesn't use the Disconnect list by default. Here is  its [the default setup](https://github.com/gorhill/uBlock/wiki/Blocking-mode:-easy-mode)

**Comment by Thorin-Oakenpants** [comment]
thanks roxkilly : check the pretty picture above. My question now would be, is there ANYTHING covered in TP that is not covered by uBo's default?

**Comment by RoxKilly** [comment]
Yes. I know this because as I pointed out in the OP:

> In some cases, my uBO filter lists and settings let something through the cracks and TP actually catches it. This is usually a tracking image of some sort

I don't use the Disconnect list in uBO. These occurrences are rare though; the next time I run across one, if I remember, I'll post it here

**Comment by crssi** [comment]
Those I have additionaly to uBo standard for TP:
`https://raw.githubusercontent.com/quidsup/notrack/master/trackers.txt`
`https://raw.githubusercontent.com/piperun/iploggerfilter/master/filterlist`
`https://raw.githubusercontent.com/metaphoricgiraffe/tracking-filters/master/trackingfilters.txt`
`https://filters.adtidy.org/extension/chromium/filters/3.txt`

I don't go FB, but to whome who does:
`https://easylist-downloads.adblockplus.org/message_seen_remover_for_facebook.txt`

I have also other collections for SB and others, if anyone interested.
Maybe this (uBo usage sharing information) should go into separate topic?

**Comment by Gitoffthelawn** [comment]
@Thorin-Oakenpants wrote
> thanks roxkilly : check the pretty picture above. My question now would be, is there ANYTHING covered in TP that is not covered by uBo's default?

> I just checked these and updated:
> Basic tracking list by Disconnect​ - 29 used out of 34
> Malvertising filter list by Disconnect - ​​​​​639 used out of 5,334

Thanks. On the surface, those results indicate that much is covered by Disconnect (and hence possibly TP) that is *not* covered by the default uBo setup.

I'm not sure if the analysis is so simple, however.  Wouldn't it depend on which **order** uBo processes lists?  IOW, if 2 lists have `foobar.com`, then I think uBo would count the item as "used" in one list and not the other.

**Comment by Thorin-Oakenpants** [comment]
@Gitoffthelawn Oh, I'm not worried about the x used out of y numbers as indicative that a ton of items are not used. I think it may actually answer the question, that 29 are not covered ("Basic tracking list by Disconnect​ - 29 used out of 34") etc.

**Comment by Thorin-Oakenpants** [comment]
There's also a Malware filter list by Disconnect​ 188 used out of 2,598

If you check the those three disconnect items, apply them/update (you can uncheck em and apply later), then the lists are available to view in your browser - since they are local

**Comment by crssi** [comment]
What I really miss in uBo is statistics... but I understand why those can't really be implemented. If those would be implemented the uBo would take a lot of CPU resources to do so.
What I mean by stats is that uBo would have a counter per list, how many hits overall and how many are unique to the specific list.
With that info we could easily decide, which lists are really needed and which are just sitting there doing nothing.

Crap, this comment might be just a spam here and should go to uBo "issue" instead.

**Comment by crssi** [comment]
Under Malware filter list by Disconnect I have 2 used out of 2598, checked now.
Two days ago I had 0 used out of XXX.
But I don't use this lists for obvious reason as they are covered by some other lists I am using.

**Comment by Thorin-Oakenpants** [comment]
^^ that indicates to me that this is what you were after. A count of which lists are being used / are most effective

**Comment by earthlng** [comment]
I think what @crssi means is a counter about which filters (and from which list) you actually encounter in your browsing.

**Comment by crssi** [comment]
Doh... my English again. :)

Yes, @earthlng, you are correct.

**Comment by Gitoffthelawn** [comment]
I was thinking about this issue a bit more, and a strong argument to keep TP enabled is that its list can be automatically updated multiple times per day (I believe).

To the contrary, most uBo lists are only automatically updated every 3 days, some only get automatically updated every 7 days, and I believe some may be scheduled to take even longer. 

Yes, you can manually force uBo updates more frequently, but it's not really reasonable to expect users to all do that.

**Comment by earthlng** [comment]
the list rarely ever [changes](https://github.com/mozilla-services/shavar-prod-lists/commits/master/disconnect-blacklist.json) though

**Comment by Gitoffthelawn** [comment]
@earthlng I tend to agree.  And at the same time, I tend to disagree. :)

I agree because when the frequency of updates is infrequent, then looking for updates frequently isn't productive.

I disagree because it doesn't take any effort from the user, and only the most severely bandwidth-restricted users will notice.  Also, sometimes a really invasive tracker can come along and needs to be blocked ASAP.  That, as you pointed out, is relatively rare.

Now, when it comes to SB (vs TB), I am starting to think it's really important to have blocklists updated more frequently than every 3 days.  And SB provides that.  From what I can tell, SB would have blocked the phishing scam described in https://www.ghacks.net/2017/05/04/fell-prey-to-google-docs-phishing-scam-do-this/ but uBo would not have blocked it for a majority of its users.

**Comment by RoxKilly** [comment]
I wrote [earlier](https://github.com/ghacksuserjs/ghacks-user.js/issues/102#issuecomment-298845122):
> In some cases, my uBO filter lists and settings let something through the cracks and TP actually catches it. This is usually a tracking image of some sort...I don't use the Disconnect list in uBO, though I have EasyList, Peter Lowe's, and EasyPrivacy ON. These occurrences are rare though; the next time I run across one, if I remember, I'll post it here

I got a hit. Turn on Tracking Protection then open [this page](https://smile.amazon.com/gp/help/customer/display.html/) with default uBO settings. uBO would not protect the user, but the built-in TP does.

**Comment by earthlng** [comment]
> I got a hit.

Hmmm? sorry can't replicate. Fresh profile with default uBO and this user.js but with TP enabled: "No tracking elements detected on this page". What isn't blocked that should be blocked?
And where did you read that we wrote to use uBO with its default settings btw?

**Comment by Thorin-Oakenpants** [comment]
> And who said to use uBO with its default settings

It's a reasonable assumption/test-case for seeing if TP actually covers something uBo doesn't. Out-of-the-box is the only way to make any decent comparison. When talking about lists earlier, here and in #103 , we have always referred to uBo defaults. I myself have specifically said so.

**Comment by RoxKilly** [comment]
@earthlng wrote:
> Can't replicate. Fresh profile with default uBO and this user.js but with TP enabled: "No tracking elements detected on this page". What isn't blocked that should be blocked?

### To replicate:
1. start with fresh Firefox 53 profile
2. Install uBlock Origin Add-On version 1.12.1 (current as of May 5 2017)
3. In `uBO settings >> 3rd-party filters`, click "Update now" to downlod latest default lists
4. Download the latest `ghacks-user.js` [live master](https://github.com/ghacksuserjs/ghacks-user.js/blob/master/user.js) and place it in Firefox profile folder
5. Change pref 0420 to `user_pref("privacy.trackingprotection.enabled", true);`
6. Comment out the two prefs under 0410d because tracking protection relies on the safebrowsing API (maybe you forgot this step?)
```
//user_pref("browser.safebrowsing.provider.mozilla.gethashURL", "");
//user_pref("browser.safebrowsing.provider.mozilla.updateURL", ""); 
```
7. Save the file and restart the browser.
8. Visit [this page](https://smile.amazon.com/gp/help/customer/display.html/) and you should get a Tracking Protection hit. The specific resource blocked is logged to the console.

If I did not have TP enabled, I would not have avoided this particular tracking mechanism. I get such a hit once or twice a week I think. If you don't get the same result, I suspect it's either because you skipped step 6 above or the Amazon server doesn't send the tracking bug to your PC, or your uBO settings aren't the default.

@earthlng wrote:
> "Google has blocked the account in the meantime, removed the fake pages, and pushed updates to Safe Browsing on top of all that."
I'm reading that as SB did NOT protect users. It would be interesting to know what updates they pushed, because I don't really understand atm what exactly the problem was.

You guys were discussing the benefits of the frequent update check by the SafeBrowsing engine as an advantage over uBO malware lists.  I think the point @Gitoffthelawn made is that because Google pushed a fix to SB infrastructure within an hour of the exploit being public, and since SB checks for updates every hour, people who use SB for protection would have had a much shorter vulnerable window than people who use just uBO with this `user.js` (they would have been exposed until their next manual update I think).

**Comment by Gitoffthelawn** [comment]
@RoxKilly wrote
>You guys were discussing the benefits of the frequent update check by the SafeBrowsing engine as an advantage over uBO malware lists. I think the point @Gitoffthelawn made is that because Google pushed a fix to SB infrastructure within an hour of the exploit being public, and since SB checks for updates every hour, people who use SB for protection would have had a much shorter vulnerable window (the quick update would protect most people who actually receive and eventually open the phishing email) than people who use just uBO with this user.js (they would have been exposed until their next manual update I think).

Exactly.  Your paraphrase of what I wrote made it much more clear though. :)

I think uBo would benefit from the ability to customize the list check interval.  Actually, do you know if it checks lists for updates or always just downloads the most recent versions?

The downside, of course, is the additional traffic if too many people start asking for lists to be updated too frequently.. it would amount to a DDOS.  But the number of people that would actually adjust such a setting would likely be minimal compared to the installed base.

Are the SB and TP lists available anywhere in a format compatible with uBo?

**Comment by earthlng** [comment]
@RoxKilly wrote

> Comment out the two prefs under 0410d because tracking protection relies on the safebrowsing API (maybe you forgot this step?)

I _did_ forget that, sorry. Ok so `amazoncustomerservice.d2.sc.omtrdc.net` is the tracker in this case.
`omtrdc.net` is listed in the Adobe section of the TP list, and also in `Dan Pollock’s hosts file` in uBo/uM.
So if you use more than only the default settings in uBO, OR also use uMatrix with its default settings, nothing slips through the cracks (for that site at least).

@Gitoffthelawn wrote

> I think uBo would benefit from the ability to customize the list check interval.

https://github.com/gorhill/uBlock/wiki/Advanced-settings

> Actually, do you know if it checks lists for updates or always just downloads the most recent versions?

afaik it checks a checksum file from gorhill's github repo before updating lists.

> Are the SB and TP lists available anywhere in a format compatible with uBo?

Not that I know. The TP list could however easily be parsed and converted into a compatible format. idk if the same is possible for SB. Would this be the format we want ... ?
```
## AddThis
||addthis.com^$third-party
||addthiscdn.com^$third-party
||addthisedge.com^$third-party
```

**Comment by Atavic** [comment]
FYI [These lists](https://github.com/ghacksuserjs/ghacks-user.js/issues/102#issuecomment-298741810) have been implemented into Ubo Lists: https://github.com/chrisaljoudi/uBlock/issues/1406#issuecomment-105517545

**Comment by RoxKilly** [comment]
@Gitoffthelawn wrote:
> Actually, do you know if it checks lists for updates or always just downloads the most recent versions?

@earthlng wrote:
> afaik it checks a checksum file from gorhill's github repo before updating lists.

That may no longer be the case. I just came across this [statement from gorhill](https://github.com/gorhill/uBlock/pull/2314#issue-201106730) from Jan 2017
> uBO no longer uses checksums.txt resource hosted on GitHub to find out whether some specific assets have changed: the update logic is now completely time-based --  checksums.txt will be deprecated and will no longer be updated. Eventually in some future it will be removed from the repository

**Comment by crssi** [comment]
Interestingly this very page/tracker is explicitly allowed on the following lists:
EasyPrivacy (under uBo privacy)
[English filter](https://filters.adtidy.org/extension/chromium/filters/2.txt)
[Spyware filter](https://filters.adtidy.org/extension/chromium/filters/3.txt)

**Comment by Gitoffthelawn** [comment]
Hmmm... Popular lists such as EasyList have a string at the top to indicate how often to perform updates.  For example: `! Expires: 4 days (update frequency)`.

IIRC, popular blocking programs like AdBlock Plus and uBlock Origin (uBo) will not automatically update the list from its source until that time has expired.

IIRC, when a user of AdBlock Plus manually updates a list, it will force an update, overriding the value specified in the header.  IIRC, uBlock Origin will not update the list in this situation, even if the user performs a manual update.

IIRC, in uBo, even if the user specifies a lower value by using the `autoUpdatePeriod` advanced setting, it will still ***not*** override the expiration interval.  Thus, it will not perform a manual update.

In uBo the user can purge all uBo caches and then perform a manual update of every list.  AFAIK, this will manually update the lists.  AFAIK, in uBo, there is no obvious way to manually purge and update a single list before it has expired, which can be days.

That's a lot of IIRC and AFAIK, so you may want to confirm.

**Comment by Thorin-Oakenpants** [comment]
Closing this. I have reached a decision. See #103 - a commit will be forthcoming in the next few days

**Comment by Thorin-Oakenpants** [comment]
https://github.com/ghacksuserjs/ghacks-user.js/commit/eaaecd128c4d646e54027712eb325e3a7d488504
- all SB enabled (i.e prefs not active and therefore do not disable it) except reporting, and real time google binary checks

https://github.com/ghacksuserjs/ghacks-user.js/commit/2a609fc85d60a73d289bf8e95383fbcf43160e8f
- all TB enabled (i.e prefs not active and therefore do not disable it) except I have enforced TP in all windows, not just PB Mode
- added basic vs strict TP block list info
- flash block list disabling (0440) also made inactive

**Comment by RoxKilly** [comment]
@Thorin-Oakenpants 

Consider writing out `SB >> SafeBrowsing` and `TP >> Tracking Protection` in your last post, because it's one of reference (almost like documentation). Also you've mistakenly used `TB` where you meant `TP`. You've done it a couple of times in this thread or in the VOTE thread, but on this last post I think it's important to get it right. It might be linked to a few times in the future and people who come directly to it may wonder what `SB`, `TP` and `TB` are until they scroll up etc...

Feel free to delete this post once you're done with it.

**Comment by Thorin-Oakenpants** [comment]
I double checked the user.js (which was what I thought you meant) and there are no `TB` 's used by mistake. I thought you may mean to expand the acronyms in the user.js. Anyway. I will scour this thread and the vote one and correct any of those typos.
